lib/include/jxl/decode.h

```diff
@@ -113,9 +113,9 @@ JXL_EXPORT void JxlDecoderDestroy(JxlDecoder* dec);
 
 /**
  * Return value for JxlDecoderProcessInput.
- * The values above 0x40 are optional informal events that can be subscribed to,
- * they are never returned if they have not been registered with
- * JxlDecoderSubscribeEvents.
+ * The values from JXL_DEC_EXTENSIONS onwards are optional informal events
```

Comment:
It was already there, but I think the word "informal" is not correct (this is not about formal versus informal). In the JxlDecoderSubscribeEvents documentation, these are called "informational", which is not the same word, and also not really the best word (all return values are in some sense informational; the word is too vague). In the comments above the enum values themselves they are called "informative" events, which is yet another word.

Also the word "optional" is not really conveying the meaning well - I think it's more that you have to "opt in" to them, but that is not really just an optional thing to do, at least if you want to get an image decoded it's actually required, not optional.

I think the whole concept could be explained a bit better, namely that there are two different kinds of return values, some that you can always get (the ones below 0x40), and others that you can only get if you subscribe to them, and most of all: that the decoder will do different things depending on what you subscribe to: if you don't subscribe to getting metadata boxes, they will just be ignored; if you don't subscribe to getting pixel data, no actual decoding will happen, etc. So subscribing to events is not just about the granularity or frequency of the information/interruptions you get during decode, but it is also about indicating what exactly you want the decoder to do.

Maybe the wording "opt-in events" is clearer than "optional informal events" or "informative events" or "informational events"?

Reply:
Also: JXL_DEC_BASIC_INFO is the first opt-in event, i.e. the "above 0x40" was meant inclusively.

Reply:
Will be going with "informative" here for now, since this is what the other text is using, but agree that this perhaps should be re-worded throughout.